controllers/drplacementcontrol_controller.go

```diff
@@ -686,20 +743,25 @@ func (d *DRPCInstance) runInitialDeployment() (bool, error) {
 // 8. Delete VRG MW from preferredCluster once the VRG state has changed to Secondary
 //
 func (d *DRPCInstance) runFailover() (bool, error) {
-	d.log.Info("Entering runFailover", "state", d.instance.Status.LastKnownDRState)
+	d.log.Info("Entering runFailover", "state", d.getLastDRState())
 
 	const done = true
+
+	if d.isFailbackInProgress() || d.isRelocationInProgress() {
+		return done, fmt.Errorf("invalid state %s for the selected action %v",
+			d.getLastDRState(), d.instance.Spec.Action)
+	}
+
```

Comment:
DRPC is relying on the status it stored earlier. This means that we need to worry about crash recovery of the operator pod after the state changed but before the state change was persisted.  I would like to see a debug assertion that the current live state (computed by querying the world now) matches the stored state.